frontend/CONTRIBUTING.md

```diff
@@ -0,0 +1,348 @@
+
+### fnm
+
+```bash
+fnm install 22.14.0
+fnm use 22.14.0
+```
+
+### nvm
+
+```bash
+nvm install 22.14.0
+nvm use 22.14.0
+```
+
+## Manage dev environment with npm
+
+### Pre-requisites
+
+Clone the Kubeflow Pipelines repo:
+
+```bash
+# Set this to your working directory
+WORKING_DIRECTORY= ...
+git clone https://github.com/kubeflow/pipelines.git ${WORKING_DIRECTORY}
+```
+
+Navigate to frontend folder:
+
+```bash
+cd ${WORKING_DIRECTORY}/frontend
+```
+
+Install the NPM dependencies:
+
+`npm ci`.
+
+`npm ci` ensures exact dependency versions are installed according to `package-lock.json`. You should run it:
+- When first setting up the project
+- After `package.json` or `package-lock.json` changes
+- In CI/CD pipelines
+
+> [!Note]
+> This command deletes the existing `node_modules` folder and requires a `package-lock.json` file.
+
+### Daily workflow
+
+You will see a lot of `npm run xxx` commands in the instructions below, the actual script being run is defined in the "scripts" field of [package.json](https://github.com/kubeflow/pipelines/blob/91db95a601fa7fffcb670cb744a5dcaeb08290ae/frontend/package.json#L32). Development common scripts are maintained in package.json, and we use npm to call them conveniently.
+
+### npm next step
+
+You can learn more about npm in https://docs.npmjs.com/about-npm/.
+
+## Start frontend development server
+
+You can then do `npm start` to run a webpack dev server at port 3000 that
+watches the source files. It also redirects api requests to localhost:3001. For
+example, requesting the pipelines page sends a fetch request to
+http://localhost:3000/apis/v1beta1/pipelines, which is proxied by the
+webserver to http://localhost:3001/apis/v1beta1/pipelines,
+which should return the list of pipelines.
+
+Follow the next section to start an api mock/proxy server to let localhost:3001
+respond to api requests.
+
+## Start api mock/proxy server
+
+### Api mock server
+
+This is the easiest way to start developing, but it does not support all apis during
+development.
+
+Run `npm run mock:api` to start a mock backend api server handler so it can
+serve basic api calls with mock data.
+
+If you want to port real MLMD store to be used for mock backend scenario, you can run the following command. Note that a mock MLMD store doesn't exist yet.
+
+```
+kubectl port-forward svc/metadata-envoy-service 9090:9090
+```
+
+### Proxy to a real cluster
+
+You can proxy requests from the UI running on your host machine to an actual KFP
+deployment running on a remote or local Kubernetes cluster. This dramatically
+improves iteration time, especially since the docker build can take 20+ minutes.
+
+KFP can be deployed in single-user or multi-user mode. Since there's a delta in
+logic between between the two modes, automated tests and manual validation
+against a single-user cluster can still fail when deployed to a multi-user
+cluster.
+
+#### Single-user
+
+1. Follow the [single-user KFP installation
```

Comment:
Do you want to mention the new Make target to setup a Kind cluster here?

Reply:
done!